zstacklib/zstacklib/utils/fileUploader.py

```
@@ -0,0 +1,289 @@
+    def _prepare_upload(self, cmd):
+        class ImageUploadDaemon(plugin.TaskDaemon):
+            def __init__(self, task):
+                super(ImageUploadDaemon, self).__init__(cmd, 'imageUpload')
+                self.task = task
+                self.task.close = self.close
+
+            def _cancel(self):
+                if self.task.completed:
+                    return
+                self.task.lastError = "image [uuid: %s] upload canceled" % cmd.imageUuid
+                shell.run('rbd rm %s' % task.tmpPath)
```

Ruff (0.13.1)

`shell` may be undefined, or defined from star imports

(F405)